src/components/Calculator.js

```diff
@@ -1,37 +1,78 @@
-const CalcBody = () => (
-  <section>
-    <div className="container">
-      <div className="button">0</div>
-      <div className="calculator-left">
-        <div className="button">AC</div>
-        <div className="button">+/-</div>
-        <div className="button">%</div>
-        <div className="button">7</div>
-        <div className="button">8</div>
-        <div className="button">9</div>
-        <div className="button">4</div>
-        <div className="button">5</div>
-        <div className="button">6</div>
-        <div className="button">1</div>
-        <div className="button">2</div>
-        <div className="button">3</div>
-        <div className="button">0</div>
-        <div className="button">.</div>
-      </div>
-      <div className="calculator-right">
-        <div className="operator">+</div>
-        <div className="operator">x</div>
-        <div className="operator">-</div>
-        <div className="operator">+</div>
-        <div className="operator">=</div>
+import React, { useEffect, useState } from 'react';
+import calculate from '../logic/calculate';
+
+function CalcFun() {
+  const [total, setTotal] = useState(0);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  useEffect(() => {
+    const handleClick = (event) => {
+      const res = calculate(
+        {
+          total,
+          next,
+          operation,
+        }, event.target.innerText,
+      );
+      setNext(res.next);
+      setTotal(res.total);
+      setOperation(res.operation);
+    };
```

Comment:
- [ ] I don't think you need to put the function in a useEffect because it needs to be accessed outside it so I will encourage you to declare it outside useEffect. Also, this is an event milestone, you need to add an event and pass the function in the event.


### Hint
You can use onClick event, you can read more on it [here](https://www.w3schools.com/react/react_events.asp)

Reply:
This change has been implemented well